runtime/binding-kafka/src/main/java/io/aklivity/zilla/runtime/binding/kafka/internal/cache/KafkaCacheCursorFactory.java

```diff
@@ -1127,35 +1145,37 @@ private KafkaFilterCondition asNotCondition(
         switch (condition.kind())
         {
         case KafkaConditionFW.KIND_KEY:
-            filterCondition = new KafkaFilterCondition.Not(asKeyCondition(condition.key()));
+            filterCondition = new KafkaFilterCondition.Not(mask, asKeyCondition(mask, condition.key()));
             break;
         case KafkaConditionFW.KIND_HEADER:
-            filterCondition = new KafkaFilterCondition.Not(asHeaderCondition(condition.header()));
+            filterCondition = new KafkaFilterCondition.Not(mask, asHeaderCondition(mask, condition.header()));
             break;
         case KafkaConditionFW.KIND_NOT:
-            filterCondition = asCondition(condition.not().condition());
+            filterCondition = asCondition(mask, condition.not().condition());
             break;
         case KafkaConditionFW.KIND_HEADERS:
-            filterCondition = new KafkaFilterCondition.Not(asHeadersCondition(condition.headers()));
+            filterCondition = new KafkaFilterCondition.Not(mask, asHeadersCondition(mask, condition.headers()));
             break;
         }
         return filterCondition;
     }
 
     private KafkaFilterCondition asHeadersCondition(
+        long mask,
         KafkaHeadersFW headers)
     {
-        return new KafkaFilterCondition.HeaderSequence(checksum, valueMatchRO, headerRO, headers);
+        return new KafkaFilterCondition.HeaderSequence(mask, checksum, valueMatchRO, headerRO, headers);
     }
 
     private static KafkaFilterCondition.Key initNullKeyInfo(
+        long mask,
```

Comment:
Before adding the `mask`, we used this as a shared instance for all `null` key conditions, but now we need the `mask` to be correct per client, so the shared part should shift to the `nullKeyRO` and we now need to create the mask-specific `Key` instance.